internal/pkg/warewulfd/warewulfd.go

```diff
@@ -14,6 +15,21 @@ import (
 
 // TODO: https://github.com/danderson/netboot/blob/master/pixiecore/dhcp.go
 // TODO: https://github.com/pin/tftp
+/*
+wrapper type for the server mux as him requests http://efiboot//grub.efi which is filtered out by http to `301 Moved Permanently` which
+which shim.fi can't handle. So filter out `//` before they hit http
+*/
+type slashFix struct {
+	mux http.Handler
+}
+
+/*
+Filter out the '//'
+*/
+func (h *slashFix) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	r.URL.Path = strings.Replace(r.URL.Path, "//", "/", -1)
+	h.mux.ServeHTTP(w, r)
+}
```

Comment:
This is probably fine; but maybe it should only filter out for calls to `http://efiboot//`?